kernel/src/net/socket/netlink/route/kernel/addr.rs

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Handle address-related requests.
+
+use core::num::NonZeroU32;
+
+use super::util::finish_response;
+use crate::{
+    net::{
+        iface::{Iface, IFACES},
+        socket::netlink::{
+            message::{CMsgSegHdr, CSegmentType, GetRequestFlags, SegHdrCommonFlags},
+            route::message::{
+                AddrAttr, AddrMessageFlags, AddrSegment, AddrSegmentBody, RtScope, RtnlSegment,
+            },
+        },
+    },
+    prelude::*,
+    util::net::CSocketAddrFamily,
+};
+
+pub fn do_get_addr(request_segment: &AddrSegment) -> Result<Vec<RtnlSegment>> {
+    let dump_all = {
+        let flags = GetRequestFlags::from_bits_truncate(request_segment.header().flags);
+        flags.contains(GetRequestFlags::DUMP)
+    };
+
+    if !dump_all {
+        return_errno_with_message!(Errno::EOPNOTSUPP, "GETADDR only supports dump requests");
+    }
+
+    let ifaces = IFACES.get().unwrap();
+    let mut response_segments: Vec<RtnlSegment> = ifaces
+        .iter()
+        // GETADDR only supports dump mode, so we're going to report all addresses.
+        .filter_map(|iface| iface_to_new_addr(request_segment.header(), iface))
+        .map(RtnlSegment::NewAddr)
+        .collect();
+
+    if response_segments.is_empty() {
+        // TDDO: Should we return an error if no addresses are found?
```

Comment:
Actually if you create an empty network space, this branch will be triggered. I'm not sure why you say it won't (the loopback interface is disabled by default so it does not have an address associated):
```
sendmsg(3, {msg_name={sa_family=AF_NETLINK, nl_pid=0, nl_groups=00000000}, msg_namelen=12, msg_iov=[{iov_base=[{nlmsg_len=24, nlmsg_type=RTM_GETADDR, nlmsg_flags=NLM_F_REQUEST|NLM_F_DUMP, nlmsg_seq=1, nlmsg_pid=0}, {ifa_family=AF_UNSPEC, ifa_prefixlen=0, ifa_flags=0, ifa_scope=RT_SCOPE_UNIVERSE, ifa_index=9999}], iov_len=24}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, 0) = 24

recvfrom(3, [{nlmsg_len=20, nlmsg_type=NLMSG_DONE, nlmsg_flags=NLM_F_MULTI, nlmsg_seq=1, nlmsg_pid=256297}, 0], 8192, 0, NULL, NULL) = 20
```

So I guess it's fine to remove these TODOs.